SimDataFormats/GeneratorProducts/interface/GenRunInfoProduct.h

```diff
@@ -68,6 +74,8 @@ class GenRunInfoProduct {
 	XSec	internalXSec_;	// the one computed during cmsRun
 	XSec	externalXSecLO_, externalXSecNLO_;	// from config file
 	double	externalFilterEfficiency_;		// from config file
+	std::vector<std::string> WeightList_;		// storing the string information corresponding to sys. weights
+	int nWeights_;					// number of weights
```

Comment:
why does nWeights need to be stored separately given that the vector stores its own size?